sunpy/cm/cm.py

```diff
@@ -156,7 +166,12 @@ def show_colormaps():
     ----------
 
     """
-    maps = sorted(cmlist)
+
+    if mission:
+        maps =  sorted({k:v for (k,v) in cmlist.iteritems() if k.count(mission)})
+    else:
```

Comment:
would it be worth returning an error if the search string did not find any entries that satisfied the search?